src/gui/qgsprojectionselector.cpp

```diff
@@ -718,6 +719,24 @@ void QgsProjectionSelector::on_lstCoordinateSystems_currentItemChanged( QTreeWid
   }
 }
 
+void QgsProjectionSelector::on_lstCoordinateSystems_itemDoubleClicked( QTreeWidgetItem *current, int column )
+{
+  Q_UNUSED( column );
+
+  QgsDebugMsg( "Entered." );
+
+  if ( !current )
+  {
+    QgsDebugMsg( "no current item" );
+    return;
+  }
+
+  // If the item has children, it's not an end node in the tree, and
+  // hence is just a grouping thingy, not an actual CRS.
+  if ( current->childCount() == 0 )
+    emit projectionDoubleClicked();
```

Comment:
The projection selector is an embedded widget, which is not always a child of a dialog (nor would it always be correct to do accept() on double click, think for e.g. of the selector in the project properties window).